### DISCUSSION TITLE
Docker container talks to service container

### DISCUSSION THREAD

**Discussion by mfridman** [community/community#26094] [Actions]
<p>Hi,</p>
<p>I have a service container (postgres) running. As outlined in <a href="https://github.com/actions/example-services/blob/989ef69ed164330bee413f11ce9332d76f943af7/.github/workflows/postgres-service.yml" rel="nofollow noopener">workflows/postgres-service.yml</a></p>
<p>The job and all the steps are running on a Linux Runner (not inside a docker container). But, I’d like to invoke a docker container on the runner to run a command <em>inside</em> a container talking to the service container.</p>
<p>E.g.,</p>
<pre><code class="lang-auto">steps:
  - name: test postgres connection
    run: psql -h localhost -d postgres -p 5432 -U user -t -c 'SELECT datname FROM pg_database;'
</code></pre>
<p>This works. But now I’d like to run the exact same command from within a docker container targeting the service postgres container:</p>
<pre><code class="lang-auto">steps:
  - name: test docker connection
    run: docker run --rm postgres:11.8-alpine sh -c "exec psql -h postgres -d postgres -p 5432 -U user -t -c 'SELECT datname FROM pg_database;"

# fails
# Error: raw error: psql: could not translate host name "postgres" to address: Name does not resolve
</code></pre>
<p>This fails, because the 2 containers are on different networks.</p>
<p>Is it possible to run the jobs on Linux Runner (not container) while invoking a docker container with <code>docker run</code> and having said container talk to the service container?</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>Based on how Docker networking works: You’d just need to connect your container to the same network as the service containers. According to documentation the name of the network is available in the <a href="https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#job-context"><code>job</code> context</a> as <code>job.container.network</code>, so something like this should work:</p>
<pre><code class="lang-yaml">steps:
  - name: test docker connection
    run: docker run --rm --network ${{ job.container.network }} postgres:11.8-alpine sh -c "exec psql -h postgres -d postgres -p 5432 -U user -t -c 'SELECT datname FROM pg_database;"
</code></pre>
<p>My guess is that they use a custom network because Docker doesn’t provide DNS on the default network.</p>

**Comment by mfridman** [comment]
<aside class="quote no-group" data-username="airtower-luna" data-post="2" data-topic="127649">
<div class="title">
<div class="quote-controls"></div>
 airtower-luna:</div>
<blockquote>
<p><code>${{ job.container.network }}</code></p>
</blockquote>
</aside>
<p>Awesome, <code>${{ job.container.network }}</code> is what I was missing. Need to spend a bit more time with the docs.</p>
<p>Indeed each gets a unique docker network.</p>
<pre><code class="lang-auto"> /usr/bin/docker network create --label 8118cb github_network_2ebe68e5c753469eb53b397da5c04539

# where 2ebe68e5c753469eb53b397da5c04539 is some unique hash every time
</code></pre>
<p>thank you <a class="mention" href="/airtower-luna">@airtower-luna</a></p>

**Comment by danstarns** [comment]
adding  --network ${{ job.container.network }} worked when manually running a docker container inside a step dependent on a service started by GitHub actions.
